### PR TITLE
🎁 1718 Use device requests API

### PR DESCRIPTION
**Pull request by mcgordonite**
Use the device requests API for:
- execute_resource
- set_resource_value

I left the now-unused `fix_path` parameter in to keep the same signature for backwards compatibility.

### CODE REVIEW

File: src/mbed_cloud/connect/connect.py [hunk 1]
```diff
@@ -348,67 +349,55 @@ def set_resource_value_async(self, device_id, resource_path,
...
-        resp = api.execute_or_create_resource(device_id,
-                                              resource_path,
-                                              **kwargs)
-        consumer = AsyncConsumer(resp.async_response_id, self._db)
-        return consumer.wait(timeout)
+        return self.execute_resource_async(device_id, resource_path).wait(timeout)
 
     @catch_exceptions(mds.rest.ApiException)
-    def execute_resource_async(self, device_id, resource_path, fix_path=True, **kwargs):
+    def execute_resource_async(self, device_id, resource_path):
```
Comment (madchutney) [inline src/mbed_cloud/connect/connect.py]: Removing the `fix_path` is definitely a breaking change, can you document it in the same way as you did for `set_resource_value`.

Removing the `kwargs` is more debatable as they were never documented.

Comment (mcgordonite) [reply src/mbed_cloud/connect/connect.py]: Good point. I've restored the fix_path parameter for both `execute_resource_async` and `set_resource_value_async`.

**Comment by alexl0gan** [comment]
Be aware, if fixed path does what I think it does, it removes the leading / for old endpoints. However DeviceRequests requires a leading / to work.

**Comment by mcgordonite** [comment]
@alexl0gan Ah, so that's a breaking change: someone could currently be passing URIs like `100/1/1` to one of these functions. So I need to check if the path starts with "/" and if not prepend it.

However, I can't make the fix_path parameter control this behaviour since that's also a breaking change.

**Comment by mcgordonite** [comment]
Looks like the base64 encoding is broken in python 2. I'll try and find a version-independent way of doing it.

**Comment by mcgordonite** [comment]
Fixed base64 encoding in Python 2.

**Pull request merged**